seahorn/lib/klee_byte_buf_helper.c

```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ */
+
+#include <limits.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <aws/common/common.h>
+#include <seahorn/seahorn.h>
+
+#include <bounds.h>
+#include <byte_buf_helper.h>
+#include <nondet.h>
+#include <proof_allocators.h>
+
+void initialize_byte_buf(struct aws_byte_buf *const buf) {
+    size_t cap = nd_size_t();
+    assume(cap <= sea_max_buffer_size());
+    buf->buffer = can_fail_malloc(cap * sizeof(*(buf->buffer)));
+    if (buf->buffer) {
+        size_t len = nd_size_t();
+        assume(len <= cap);
+        buf->len = len;
+        buf->capacity = cap;
+    }
+    else {
+        buf->len = 0;
+        buf->capacity = 0;
+    }
+    buf->allocator = sea_allocator();
+}
+
+void initialize_byte_cursor(struct aws_byte_cursor *const cursor) {
+    size_t max_buffer_size = sea_max_buffer_size();
+    cursor->ptr = can_fail_malloc(sizeof(*(cursor->ptr)) * max_buffer_size);
+    if (cursor->ptr){
+        cursor->len = nd_size_t();
+        assume(cursor->len <= max_buffer_size);
+    }
+    else cursor->len = 0;
+}
+
+bool aws_byte_buf_is_bounded(
```

Comment:
functions from this line further can be shared between klee and seahorn and should be factored out to avoid duplication